homeassistant/components/template/config_flow.py

```diff
@@ -44,20 +44,29 @@ def generate_schema(domain: str, flow_type: str) -> dict[vol.Marker, Any]:
     """Generate schema."""
     schema: dict[vol.Marker, Any] = {}
```

Comment:
```suggestion
    schema: dict[vol.Marker, Any] = {
        vol.Required(CONF_STATE): selector.TemplateSelector(),
    }
```

Reply:
This is in preparation for adding the flow to other platforms, some that do not use CONF_STATE, so I think the suggested change is not viable as I will have to change the platform implementation PR (example #120886), so the idea is to do here to reduce the size of the other.

This goes for all the comments below...

Reply:
Reviewers are reviewing only the code of the actual PR and we omit potential future PRs from our review process as we don't know if they will ever be created or get merged. As #120886 or any other PR was not mentioned in the PR description, I reviewed only the code in this PR

Will now re-review it